### PR TITLE
[Exhaustive Search] Add search jobs list page

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/55708
Closes https://github.com/sourcegraph/sourcegraph/issues/55706

[Figma Designs](https://www.figma.com/file/H4ezuXmEvVeoC6UaiNSbEa/Exhaustive-search-UI?type=design&node-id=1577-12054&mode=design&t=fwULVv6bj9KuZf8D-0)

This PR adds a new page (note that this is behind `searchJobs` experimental feature flag). 
All newly added UI isn't connected to the API yet. It will be done in further PR as soon as end-to-end backend flow is in place.

## Screenshots
| Size | Desktop | Mobile |
| ----- | ------ | ------ | 
| List page with items | Screenshot 2023-09-05 at 22 46 05 | Screenshot 2023-09-05 at 22 46 21 | 
| Dark mode | Screenshot 2023-09-05 at 22 47 15 | Screenshot 2023-09-05 at 23 14 55 | 

## Test plan
- Enabled `searchJobs` experimental feature flag in the settings
- Navigate to the search result page through global navigation search dropdown item
- Check general layout, note that in the webapp we don't have any mocks so there won't be any search jobs items
- Run `pnpm storybook:web` command in the sourcegraph repository root 
- Go to the `http://localhost:9001/?path=/story/web-search-jobs-searchjobspage--search-jobs-list-page` page in the browser. 
- Check layout with mocks (search jobs list page should have 5 different items) 
- Check mid-size and small screen layouts

### CODE REVIEW

**Review by fkling — APPROVED** [review]

File: client/web/src/enterprise/search-jobs/SearchJobsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+                <SyntaxHighlightedSearchQuery query={job.query} />
+            </span>
+
+            <span className={styles.jobCreator}>
+                <UserAvatar user={job.creator!} />
+                {job.creator?.displayName}
+            </span>
+
+            <span className={styles.jobActions}>
+                <Button variant="link">ViewLogs</Button>
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/enterprise/search-jobs/SearchJobsPage.tsx]: 🤦 I missed this, thanks for the catch

**Comment by vovakulikov** [comment]
@rrhyne Just to avoid having a big PR in the review, I suggest we review this one without a new zero state and I will add a new zero state in a follow-up PR.

**Comment by rrhyne** [comment]
Roger. I'll remove the old empty state screenshot.

My one issue with the storybook is, what's the difference between failed and errored? Shouldn't those be the same status? If not, shouldn't the errored be red as well? I can' think of no status we would have for 'warning', other than maybe, "this is taking a long time" but we don't really track that I bet and I don't think we need to.

**Comment by vovakulikov** [comment]
@rrhyne, good question. I had a chat with @stefanhengl about this as well earlier this week. So, tech difference is errored status does mean that some work in the search job has errored during the search, but the job in general keeps going and may produce some results at the end, in fact @stefanhengl told me that on the backend, we will retry errored parts. Failing means that something very bad happened during the search job, and the job isn't doing anything at the moment. 

So, this being said, I don't really know how to combine or present these two different states, 
Maybe if we retry errored parts on the backend, we shouldn't change the status from the progress mode and present this info about errored and rerun actions in the logs? @rrhyne wdyt?

**Comment by rrhyne** [comment]
> Maybe if we retry errored parts on the backend, we shouldn't change the status from the progress mode and present this info about errored and rerun actions in the logs? @rrhyne wdyt?

I think you all should seriously consider this as an option. I don't have time to help, but you seem to be on the right track... if we are still working on retrying and returning results, I don't think we should change status.

**Review by rrhyne — APPROVED** [review]
Looks great, thanks!

**Pull request merged**